From a393bc2f16c8de9c56d5d2a30340ca40097350e4 Mon Sep 17 00:00:00 2001
From: vladimir.a.ivanov <vladimir.a.ivanov@intel.com>
Date: Fri, 11 Oct 2013 12:44:26 +0700
Subject: Dalvik: include should be fixed in dalvik/libcrash/DbgBridge.h

BZ: 144363

Header files should be included by variable LOCAL_C_INCLUDES and path from it
instead of relative path. Fix it in DbgBridge.h.

Category: device-enablement
Domain: AOSP-Dalvik-Runtime
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I92fffe3b78ac1bb785d80aa64b7b0ade7d416ab0
Orig-MCG-Change-Id: Ic913d1b766d42f1467dd6fd7a2207e018ba1fd2a
Signed-off-by: vladimir.a.ivanov <vladimir.a.ivanov@intel.com>
Reviewed-on: http://android.intel.com:8080/137775
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Reviewed-by: Popov, Ivan G <ivan.g.popov@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: Banerji, Udayan <udayan.banerji@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 libcrash/Android.mk  |    3 ++-
 libcrash/DbgBridge.h |    2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/libcrash/Android.mk b/libcrash/Android.mk
index c64a021..9cdc0e3 100644
--- a/libcrash/Android.mk
+++ b/libcrash/Android.mk
@@ -43,6 +43,7 @@ LOCAL_C_INCLUDES += \
     dalvik/vm/interp \
     external/stlport/stlport \
     bionic \
-    bionic/libstdc++/include
+    bionic/libstdc++/include \
+    system
 
 include $(BUILD_SHARED_LIBRARY)
diff --git a/libcrash/DbgBridge.h b/libcrash/DbgBridge.h
index c6d0dc2..d28f6c4 100644
--- a/libcrash/DbgBridge.h
+++ b/libcrash/DbgBridge.h
@@ -20,7 +20,7 @@
 #define CRASH_HANDLER_BRIDGE_H_
 
 #include <sys/types.h>
-#include "../../../system/core/debuggerd/utility.h"
+#include "core/debuggerd/utility.h"
 
 /*
  * @brief Dump the Dalvik VM's specific information to the tombstone file
-- 
1.7.4.1

